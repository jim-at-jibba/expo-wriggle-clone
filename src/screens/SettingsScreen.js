import React from 'react';
import { Image, ScrollView, Text, View } from 'react-native';
import { fonts, utilStyles } from '../api/constants';
import appImages from '../api/preloadImages';

const headerImage = __DEV__ ? 'rabbitDev' : 'rabbitProd';
const headerTitle = (
  <View style={{ flex: 1 }}>
    <Image
      style={{ alignSelf: 'center', height: 40, width: 40 }}
      source={appImages[headerImage]}
    />
  </View>
);

const SettingsScreen = () => (
  <View style={utilStyles.container}>
    <ScrollView contentContainerStyle={utilStyles.contentContainer}>
      <Text style={utilStyles.text}>Settings Content Area</Text>
      <Text style={{ fontFamily: fonts.pacifico, fontSize: 20, marginTop: 16 }}>
        Pacifico Font Example
      </Text>
    </ScrollView>
  </View>
);

SettingsScreen.navigationOptions = {
  headerLeft: (
    <View style={{ flex: 1, paddingLeft: 8 }}>
      <Text>left</Text>
    </View>
  ),
  headerRight: (
    <View style={{ flex: 1, paddingRight: 8 }}>
      <Text>right</Text>
    </View>
  ),
  // title: 'Settings'
  headerTitle
};

/*
// shoutout brent: https://snack.expo.io/H105kxsG7
const shouldShowBackButton = stackRouteNavigation => {
  const parent = stackRouteNavigation.dangerouslyGetParent();
  return parent.state.routes.indexOf(stackRouteNavigation.state) > 0;
};

SettingsScreen.navigationOptions = ({ navigation }) => ({

headerLeft: !shouldShowBackButton(navigation) ? (
  <View style={{ flex: 1 }}>
    <Text>left</Text>
  </View>
) : null,
*/

export default SettingsScreen;
