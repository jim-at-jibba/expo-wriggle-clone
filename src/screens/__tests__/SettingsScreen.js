import React from 'react';
import renderer from 'react-test-renderer';

import SettingsScreen from '../SettingsScreen';

const navigation = {
  goBack: jest.fn(),
  navigate: jest.fn()
};

const tree = renderer
  .create(<SettingsScreen navigation={navigation} />)
  .toJSON();

describe('<SettingsScreen />', () => {
  it('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});
