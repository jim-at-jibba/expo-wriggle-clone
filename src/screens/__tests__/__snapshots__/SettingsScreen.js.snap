// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
/>
`;
